### PR TITLE
Add benchmark for Sobel filters

### PR DESCRIPTION
**Pull request by jni**
## Description
Supercedes #3212 

## Checklist
[It's fine to submit PRs which are a work in progress! But before they are merged, all PRs should provide:]
- [x] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- [x] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- ~~[ ] Gallery example in `./doc/examples` (new features only)~~
- [x] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- ~~[ ] Unit tests~~

[For detailed information on these and other aspects see [scikit-image contribution guidelines](http://scikit-image.org/docs/dev/contribute.html)]

## References
[If this is a bug-fix or enhancement, it closes issue # ]
[If this is a new feature, it implements the following paper: ]

## For reviewers

(Don't remove the checklist below.)

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

File: benchmarks/benchmark_filters.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import numpy as np
+from skimage import filters
+
+
+class FiltersSuite:
+    """Benchmark for filter routines in scikit-image."""
+    def setup(self):
+        self.image = np.random.random((4000, 4000))
+        self.image[:2000, :2000] += 1
+        self.image[3000:, 3000] += 0.5
```
Comment (soupault) [inline benchmarks/benchmark_filters.py]: @jni Should we stay within our dtype standards (i.e. range [-1|0, 1]) ? :)

Comment (jni) [reply benchmarks/benchmark_filters.py]: I don't think this matters.

Comment (jni) [reply benchmarks/benchmark_filters.py]: to be more precise: certainly it doesn't matter in the benchmarks. As to the library as a whole, I really want it to be friendlier to "out-of-range" images, so I don't mind not adhering to that standard where we don't have to. Of course, the overall messaging will need to change.

**Comment by JDWarner** [comment]
Approved, no changes (@stefanv beat me to the button)

**Comment by jni** [comment]
@KimNewell your code is now in scikit-image! =)

**Pull request merged**